ipalib/frontend.py

```diff
@@ -1041,18 +1043,23 @@ def output_for_cli(self, textui, output, *args, **options):
 
         self.log_messages(output)
 
-        order = [p.name for p in self.output_params()]
+        order = deque()
+        labels = dict()
+        flags = dict()
+
+        for p in self.output_params():
+            order.append(p.name)
+            labels[p.name] = unicode(p.label)
+            flags[p.name] = p.flags
+
         if options.get('all', False):
-            order.insert(0, 'dn')
+            order.appendleft('dn')
```

Comment:
Unless it's performed in a loop, one ``insert(0, item)`` is ok. A deque is overkill. It requires more memory and doesn't amortize that fast.

Reply:
Will return list then :)